src/controller/org.controller/org.controller.js

```diff
@@ -529,6 +534,11 @@ async function updateUser (req, res, next) {
       return res.status(404).json(error.userDne(username))
     }
 
+    if ((requesterUsername !== username || requesterShortName !== shortName) && !isSecretariat && !isAdmin) {
```

Comment:
Take a look at the check in lines 726-728 of the `resetSecret` function. The nested check is making sure that only users with the "Admin" role that belong to the same org as the user being updated can actually update the user. Previously, the check was missing this requirement and it allowed any user with "Admin" role to update another user. This modification prevents that.